src/blog/2024/04/node-red-dashboard-2-layout-navigation-styling.md

```diff
@@ -0,0 +1,186 @@
+
+#### Apear over content 
+
+When the sidebar is opened, the page is partially covered by a transparent layer, and the sidebar appears on top of this layer
+
+!["Image showing 'Apear over content' sidebar"](./images/node-red-dashboard-2-layout-navigation-styling-appear-over-content.gif "screenshot displaying searching for botFather bot for creating custom bot"){data-zoomable}
+
+In this type of sidebar, you can notice how the sidebar opens without affecting the width of the page content
+
+#### Always hide 
+
+In this type, the sidebar is always hidden, and navigation between different pages can be achieved using the ui-control widget.
+
+!["screenshot displaying searching for botFather bot for creating custom bot"](./images/node-red-dashboard-2-layout-navigation-styling-always-hidden.png "screenshot displaying searching for botFather bot for creating custom bot"){data-zoomable}
+
+### Setting sidebar 
+
+1. Navigate to the Dashboard 2.0 sidebar.
+2. Click on the "Edit Settings" button located at the top left side of the Dashboard 2.0 sidebar.
+3. Select your preferred sidebar style from the "Style" field in the sidebar options section.
+
+!["Image showing process of changing sidebar style"](./images/node-red-dashboard-2-layout-navigation-styling-setting-sidebar.gif "Image showing process of changing sidebar style"){data-zoomable}
+
+## Styling your Dashboard 2.0 elements
+ 
+In Dashboard 2.0, we can add classes to almost all widgets, pages, and groups and style them using CSS. Applying styles to these classes will inject the CSS inline, giving them an inline scope. This means the specified styles will take precedence over any existing CSS rules.
+
+### Adding classes
+
+1. To add classes to your widget, page, or group, you'll need to open its configuration
+2. Find the 'Class' field and enter your class.
+
+### Writing custom CSS
+
+In Dashboard 2.0, the ui-template allows you to write custom CSS for Dashboard 2.0 elements. In the ui-template, you can add CSS for two different scopes: Single page (selecting this allows you to specify CSS for all classes within the selected page) and All pages (selecting this allows you to specify CSS for classes across all pages in a ui-base).
+
+1. Drag an ui-template widget onto the canvas.
+2. Double-click on it and select the scope within the type field.
+3. If you select the single-page scope, you'll need to select the page to which your element or class belongs for adding styling. If you select the all-page scope, then you'll need to select the ui-base that includes those pages to which you want to add styling.
+4. Now you can write your custom CSS within the ui-template.
+
+!["Screenshot of ui-template widget adding CSS for styling elements of all pages of a ui-base"](./images/node-red-dashboard-2-layout-navigation-styling-adding-style-in-ui-template.png "Screenshot of ui-template widget adding CSS for styling elements of all pages of a ui-base"){data-zoomable}
```

Comment:
This would be better as a side-by-side. Left-side showing the widget/group/page on which you've defined the custom class, the right-side then showing the `ui-template` with the defined CSS